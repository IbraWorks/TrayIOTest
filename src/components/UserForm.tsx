import React, { useState } from "react";
import RegistrationSection from "./RegistrationSection";
import PrivacySection from "./PrivacySection";
import UserFormSuccess from "./UserFormSuccess";
import { IFormValues, IRegistrationValues, IPrivacyValues } from "../interfaces/IFormValues";

import { Grid, Menu } from "semantic-ui-react";

const UserForm: React.FC = () => {
  const [stage, setStage] = useState<number>(1);
  const [formValues, setFormValues] = useState<IFormValues>({
    registrationValues: {
      name: "",
      role: "",
      email: "",
      password: ""
    },
    privacyValues: {
      updatesByEmail: "",
      communicationByEmail: ""
    }
  });

  const nextStage = () => {
    setStage(stage => stage + 1);
  };

  // const previousStage = () => {
  //   setStage(stage - 1);
  // };

  const handleRegistrationData = (data: IRegistrationValues) => {
    setFormValues({...formValues, registrationValues: data});
    nextStage();
  };

  const handlePrivacyData = (data: IPrivacyValues) => {
    setFormValues({...formValues, privacyValues: data});
    nextStage();
  }

  const renderSwitchForm = (stage: number) => {
    switch (stage) {
      case 1:
        return (
          <RegistrationSection
            handleData={handleRegistrationData}
            registrationValues={formValues.registrationValues}
          />
        );
      case 2:
        return (
          <PrivacySection handleData={handlePrivacyData} privacyValues={formValues.privacyValues} />
        );
      case 3:
        return <UserFormSuccess formValues={formValues}/>;
    }
  };

  return (
    <div>
      <Menu pointing secondary widths="4">
        <Menu.Item color="blue" name="User" active={stage === 1} />
        <Menu.Item color="blue" name="Privacy" active={stage === 2} />
        <Menu.Item color="blue" name="Done" active={stage === 3} />
      </Menu>
      <Grid
        textAlign="center"
        style={{ height: "60vh" }}
        verticalAlign="middle"
      >
        <Grid.Column style={{ maxWidth: 500 }}>
          {renderSwitchForm(stage)}
        </Grid.Column>
      </Grid>
    </div>
  );
};

export default UserForm;
