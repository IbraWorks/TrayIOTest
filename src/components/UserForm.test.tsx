import { shallow, ShallowWrapper } from 'enzyme'
import UserForm from './UserForm'
import React from 'react'
import { Menu } from 'semantic-ui-react';

describe("UserForm", () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<UserForm />)
    })
    it('should render with user privacy and done headers', () => {
          expect(wrapper.exists()).toBe(true)
    });

    it('matches snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should activate User header initally', () => {
        expect(wrapper.find(Menu).find("MenuItem[name='User']").props()).toEqual({"active": true, "color": "blue", "name": "User"})
        expect(wrapper.find(Menu).find("MenuItem[name='Privacy']").props()).toEqual({"active": false, "color": "blue", "name": "Privacy"})
        expect(wrapper.find(Menu).find("MenuItem[name='Done']").props()).toEqual({"active": false, "color": "blue", "name": "Done"})
    });
})