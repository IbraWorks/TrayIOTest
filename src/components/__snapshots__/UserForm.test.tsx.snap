// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm matches snapshot 1`] = `"<div><div class=\\"ui pointing secondary four item menu\\"><div class=\\"blue active item\\">User</div><div class=\\"blue item\\">Privacy</div><div class=\\"blue item\\">Done</div></div><div style=\\"height:60vh\\" class=\\"ui center aligned middle aligned grid\\"><div style=\\"max-width:500px\\" class=\\"column\\"><div><form class=\\"ui form\\"><div class=\\"required field\\"><label style=\\"float:left\\">Name</label><input name=\\"name\\" placeholder=\\"John Smith\\" value=\\"\\"/></div><div class=\\"field\\"><label style=\\"float:left\\">Role</label><input name=\\"role\\" placeholder=\\"Software Engineer\\" value=\\"\\"/></div><div class=\\"required field\\"><label style=\\"float:left\\">Email</label><input name=\\"email\\" placeholder=\\"email@example.com\\" value=\\"\\"/></div><div class=\\"required field\\"><label style=\\"float:left\\">Password</label><input type=\\"password\\" name=\\"password\\" placeholder=\\"password\\" value=\\"\\"/></div><button style=\\"float:right\\" type=\\"submit\\" class=\\"ui green button\\">Submit</button></form></div></div></div></div>"`;
